Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/core/impl/IndexService.java

```diff
@@ -79,52 +80,58 @@
 	 * @throws IndexerRunningException
 	 */
 	@Override
-	public void insertConcept(ConceptEntry entry)
+    public void insertConcept(ConceptEntry entry, String userName)
 			throws IllegalAccessException, LuceneException, IndexerRunningException {
 		if (indexerRunningFlag.get()) {
 			throw new IndexerRunningException(indexerRunning);
 
 		}
-		luceneUtility.insertConcept(entry);
+        luceneUtility.insertConcept(entry, userName);
 	}
 
 	/**
 	 * This method deletes the concept in lucene index based on id of the
 	 * concept
 	 */
 	@Override
-	public void deleteById(String id) throws LuceneException, IndexerRunningException {
+    public void deleteById(String id, String userName) throws LuceneException, IndexerRunningException {
 		if (indexerRunningFlag.get()) {
 			throw new IndexerRunningException(indexerRunning);
 		}
-		luceneUtility.deleteById(id);
+        luceneUtility.deleteById(id, userName);
 
 	}
 
 	/**
 	 * This method deletes the index in lucene
 	 */
 	@Override
-	public void deleteIndexes() throws LuceneException, IndexerRunningException {
+    public void deleteIndexes(String userName) throws LuceneException, IndexerRunningException {
 		if (!indexerRunningFlag.compareAndSet(false, true)) {
 			throw new IndexerRunningException(indexerRunning);
 		}
-		luceneUtility.deleteIndexes();
+        luceneUtility.deleteIndexes(userName);
 		indexerRunningFlag.set(false);
 	}
 
-	/**
-	 * This method indexes concepts in lucene and runs indexer only once
-	 */
-	@Override
-	public void indexConcepts()
-			throws LuceneException, IllegalArgumentException, IllegalAccessException, IndexerRunningException {
-		if (!indexerRunningFlag.compareAndSet(false, true)) {
-			throw new IndexerRunningException(indexerRunning);
-		}
-		luceneUtility.indexConcepts();
-		indexerRunningFlag.set(false);
-	}
+    /**
+     * This method indexes concepts in lucene and runs indexer only once
+     */
+    @Override
+    @Async("indexExecutor")
+    public void indexConcepts(String userName)
+            throws LuceneException, IllegalArgumentException, IllegalAccessException, IndexerRunningException {
+        if (!indexerRunningFlag.compareAndSet(false, true)) {
+            throw new IndexerRunningException(indexerRunning);
+        }
+        try {
+            luceneUtility.indexConcepts(userName);
+        } catch (IllegalArgumentException | IllegalAccessException | LuceneException e) {
+            throw e;
```

Comment:
if you just throw the exception without processing, there is no need to catch it first.